server/src/main/java/io/crate/planner/node/ddl/CreateRolePlan.java

```diff
@@ -53,19 +69,77 @@ public StatementType type() {
+
+        Map<String, Object> properties = validate(
             createRole.properties(),
-            params,
             plannerContext.transactionContext(),
-            plannerContext.nodeContext());
+            plannerContext.nodeContext(),
+            params,
+            subQueryResults
+        );
+        SecureHash newPassword = UserActions.generateSecureHash(properties);
 
         if (createRole.isUser() == false && newPassword != null) {
             throw new UnsupportedOperationException("Creating a ROLE with a password is not allowed, " +
                                                     "use CREATE USER instead");
         }
 
-        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword)
+        // Map is always not null to simplify streaming.
+        // It's safe to create a JwtProperties object as user input already passed validation.
+        Map<String, Object> jwtPropertiesMap = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+        JwtProperties jwtProperties = null;
+        if (jwtPropertiesMap.isEmpty() == false) {
+            String iss = Maps.get(jwtPropertiesMap, "iss");
+            String username = Maps.get(jwtPropertiesMap, "username");
+            jwtProperties = new JwtProperties(iss, username);
+        }
+
+        roleManager.createRole(createRole.roleName(), createRole.isUser(), newPassword, jwtProperties)
             .whenComplete(new OneRowActionListener<>(consumer, rCount -> new Row1(rCount == null ? -1 : rCount)));
     }
+
+    public static Map<String, Object> validate(GenericProperties<Symbol> genericProperties,
+                                               TransactionContext txnCtx,
+                                               NodeContext nodeContext,
+                                               Row params,
+                                               SubQueryResults subQueryResults) {
+        Function<? super Symbol, Object> eval = x -> SymbolEvaluator.evaluate(
+            txnCtx,
+            nodeContext,
+            x,
+            params,
+            subQueryResults
+        );
+        Map<String, Object> properties = genericProperties.map(eval).properties();
+        for (var entry : properties.entrySet()) {
+            var key = entry.getKey();
+            switch (key) {
+                case JWT_PROPERTY_KEY:
+                    Map<String, Object> props = Maps.getOrDefault(properties, JWT_PROPERTY_KEY, Map.of());
+                    // Check that all required properties are specified.
+                    for (String property: JWT_PROPERTIES) {
+                        if (props.containsKey(property) == false) {
+                            throw new IllegalArgumentException(
+                                String.format(Locale.ENGLISH, "JWT property must have property '%s'", property)
+                            );
+                        }
+                    }
```

Comment:
Given that the schema is known and fixed, couldn't we already turn this into a `JwtProperties` object here via a `JwtProperties.parse` or something like that?

Reply:
https://github.com/crate/crate/pull/15515#discussion_r1479572845